### PR TITLE
[AddressLowering] Storage root inherits lexical.

### PR DESCRIPTION
**Pull request by nate-chandler**
When a `begin_borrow [lexical]` is lowered, the lifetime that it describes can't be shortened (or eliminated) when lowering.  In some cases, though, there will not be an alloc_stack corresponding directly to the value being borrowed.

In these cases, mark the whole aggregate lexical.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please smoke test macOS platform

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test macOS platform

**Comment by nate-chandler** [comment]
@swift-ci please test macOS platform

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 1]
```diff
@@ -2776,6 +2776,13 @@ void UseRewriter::visitBeginBorrowInst(BeginBorrowInst *borrow) {
       allocStack->setIsLexical();
       return;
     }
+    auto storage = AccessStorageWithBase::compute(address);
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/AddressLowering.cpp:2779]: If you're not interested in AccessStorage, you should be able to use `getAccessBase` or `AccessBase::compute`

**Review by atrick — APPROVED** [review]
Nice.

Can you add the test case in which the payload is a tuple that the pattern match logic needs to destructure?

Should we also have a .sil test case for `struct_extract`?

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp (hunk 1 above)
Comment (nate-chandler) [reply lib/SILOptimizer/Mandatory/AddressLowering.cpp]: Oh, perfect!  Changed to `getAccessBase` and checked propagated the lexical flag to it if its an `alloc_stack` and allowed it if it was a function argument.  Added a struct extract test case for both an opaque value that needed an alloc_stack and one that was passed in.

**Comment by nate-chandler** [comment]
@swift-ci please test

**Pull request merged**